File: Dart/tic_tac_toe/lib/player.dart
/// A Tic-Tac-Toe player represented by a number.
enum Player {
  empty(0),
  X(1),
  O(2);

  /// The raw value associated with this player.
  final int value;
  const Player(this.value);

  /// Alternate between [Player.X] and [Player.O].
  Player nextPlayer() {
    return this == Player.X ? Player.O : Player.X;
  }

  /// Gets an icon to use for this [Player]. Including [colorize] will add
  /// escape codes for a colorful output when printed on the command line.
  String icon({bool colorize = true}) {
    // Get the base icon to use.
    String icon;
    switch (this) {
      case Player.X:
        icon = "X";
        break;
      case Player.O:
        icon = 'O';
        break;
      default:
        icon = ' ';
    }

    // Colorize the icon if possible.
    if (colorize && this != Player.empty) {
      switch (this) {
        case Player.X:
          icon = [_cyanEscapeCode, icon, _resetEscapeCode].join();
          break;
        case Player.O:
          icon = [_magentaEscapeCode, icon, _resetEscapeCode].join();
          break;
        default:
          break;
      }
    }

    return icon;
  }
}

const _cyanEscapeCode = "\u001b[36m";
const _magentaEscapeCode = "\u001b[35m";
const _resetEscapeCode = "\u001b[0m";

File: Python/tic_tac_toe/player.py
from enum import Enum

_cyanEscapeCode = "\u001b[36m"
_magentaEscapeCode = "\u001b[35m"
_resetEscapeCode = "\u001b[0m"


class Player(Enum):
    """A Tic-Tac-Toe player represented by a number."""

    EMPTY = 0
    X = 1
    O = 2

    def nextPlayer(self) -> "Player":
        """Alternate between `Player.X` and `Player.O`."""
        return self.X if self == self.O else self.O

    def icon(self, colorize=True) -> str:
        """Gets an icon to use for this player.

        Including `colorize` will add escape codes for a
        colorful output when printed on the command line.
        """
        # Get the base icon to use.
        icon = " "
        if self == self.X:
            icon = "X"
        elif self == self.O:
            icon = "O"

        # Colorize the icon if possible.
        if colorize and self != self.EMPTY:
            if self == self.X:
                icon = _cyanEscapeCode + icon + _resetEscapeCode
            elif self == self.O:
                icon = _magentaEscapeCode + icon + _resetEscapeCode

        return icon


if __name__ == "__main__":
    player = Player.X
    print(player.icon())
